# Pontifícia Universidade Católica de São Paulo

## Bacharelado em Ciência de Dados e Inteligência Artificial
#### 3 Semestre

### Aluno: Lucas Lopes Amorim


## questão dirigida

Como examinar os efeitos de duas ou mais variáveis simultaneamente?

## contexto e motivação

Após fazer a revisão da literatura na procura por proxies da evasão escolar, Horácio necessita coletar dados para poder treinar modelos de aprendizagem de máquina. No entanto, ele ainda não se sente seguro de que um modelo baseado nestes dados pode efetivamente identificar estudantes que estejam em risco de evasão. 

Horácio decide então fazer um experimento para entender melhor os dados. Para simplificar, Horácio decidiu tentar primeiramente identificar os estudantes desmotivados, assumindo que este grupo possui maior risco de evasão. Mais precisamente, Horácio está interessado em comparar o grau de satisfação dos estudantes de quatro níveis de renda (A, B, C, D) com a motivação com o curso. Horácio aplica um questionário de 40 itens projetado para medir o grau de motivação com o curso.

A consolidação dos dados está nesta planilha: http://shorturl.at/ivFWY

Horácio estabelece as suas hipóteses:

Horácio estabelece as suas hipóteses:
- $H_0$. A renda familiar do estudante não tem relação com sua motivação com o curso
<br></br>
- $H_1$. A renda familiar do estudante tem relação com sua motivação com o curso

Como Horácio pode testar suas hipóteses?

O teste que precisa ser feito é o teste **ANOVA**

## Teste ANOVA

In [1]:
# Importando bibiotecas necessárias
import pandas as pd

In [3]:
# Carregando os dados
df = pd.read_csv('renda-e-motivação.tsv', sep='\t')
df.head()

Nível de renda  Motivação com o curso
0              D                      3
1              D                      6
2              D                      7
3              D                      2
4              D                      0

### 1. Encontre a média para cada amostra

In [8]:
means = df.groupby('Nível de renda')['Motivação com o curso'].mean().to_dict()
means

{'A': 25.6, 'B': 25.0, 'C': 14.0, 'D': 3.6}

### 2. Encontre a soma dos escores, soma dos quadrados dos escores, número total de respondetes e a média de todos os grupos

In [19]:
# Calcule a soma dos escores e a soma dos escores quadrados.
soma_escores = df['Motivação com o curso'].sum()
df['Motivação com o curso quadrado'] = df['Motivação com o curso'].apply(lambda x: x**2)
soma_escores_quadrados = df['Motivação com o curso quadrado'].sum()
soma_escores_quadrados

8099

In [20]:
# Calcule o número total de participantes da amostra e a média para todos os grupos combinados.
x_total = df['Motivação com o curso'].sum()/len(df)
x_total

17.05

### 3. Calcule a soma dos quadrados totais

In [22]:
SS_total = soma_escores_quadrados - len(df)*(x_total)**2
SS_total

2284.949999999999

### 4. Calcule a soma dos quadrados entre grupos

In [45]:
groups_stats = df.groupby('Nível de renda')['Motivação com o curso'].agg(['count', 'mean'])
SS_entre = (groups_stats['count']*groups_stats['mean']**2).sum() - len(df)*(x_total)**2
SS_entre

1632.5500000000002

### 5. Calcule a soma dos quadrados intra-grupos.

In [38]:
SS_intra = soma_escores_quadrados - (groups_stats['count']*groups_stats['mean']**2).sum()
SS_intra

652.3999999999987

### 6. Calcule os graus de liberdade entre grupos.

In [42]:
k = len(df['Nível de renda'].unique())
df_entre = k - 1
df_entre

3

### 7. Calcule os graus de liberdade intra-grupos.

In [43]:
df_intra = len(df) - k
df_intra

16

### 8. Calcule o quadrado médio entre grupos.

In [46]:
MS_entre = SS_entre/df_entre
MS_entre

544.1833333333334

### 9. Calcule o quadrado médio intra-grupos.

In [47]:
MS_intra = SS_intra/df_intra
MS_intra

40.77499999999992

### 10. Calcule a Razão F.

In [48]:
F = MS_entre/MS_intra
F

13.346004496219116

Valor do F na tabela de significância para α = 0.05 e df_entre = 3 e df_intra = 16: **3.24**

Como 13.346 >> 3.24 temos que **a classe (nível de renda) afeta** a motivação dos estudantes.